common/models/role.js

```diff
@@ -179,8 +179,10 @@ module.exports = function(Role) {
     }
     var modelClass = context.model;
     var modelId = context.modelId;
-    var userId = context.getUserId();
-    Role.isOwner(modelClass, modelId, userId, callback);
+    var user = context.getUser();
+    var userId = user && user.id;
+    var principalType = user && user.principalType;
+    Role.isOwner(modelClass, modelId, userId, principalType, callback);
   });
 
```

Comment:
now that context.getUser() can return null, we need to test user before getting user.id and user.principalType